docs/content/guides/developer/app-examples/escrow.mdx

```diff
@@ -3,4 +3,160 @@ title: Escrow
+
+Let’s create a Sui project using the terminal command `sui move new escrow`. Create a file in the sources directory named `shared.move`, and let’s go through the code together.
+
+See [this](https://github.com/MystenLabs/sui/blob/main/examples/move/escrow/sources/shared.move) for the complete code.
+
+### `shared.move`
+
+Let's go through the code line by line:
+
+#### Structs
+
+##### `Escrow<T>`
+
+This struct represents an object held in escrow. It contains the following fields:
+
+- `id`: Unique identifier for the escrow object.
+- `sender`: Address of the owner of the `escrowed` object.
+- `recipient`: Intended recipient of the `escrowed` object.
+- `exchange_key`: ID of the key that opens the lock on the object sender wants from the recipient.
+- `escrowed`: The actual object held in escrow.
+
+```move
+struct Escrow<T: key + store> has key, store {
+   id: UID,
+   sender: address,
+   recipient: address,
+   exchange_key: ID,
+   escrowed: T,
+}
+```
+
+#### Error Codes
+
+Two constants are defined to represent potential errors during the execution of the swap:
+
+- `EMismatchedSenderRecipient`: The `sender` and `recipient` of the two escrowed objects do not match.
+- `EMismatchedExchangeObject`: The `exchange_for` fields of the two escrowed objects do not match.
+
+```move
+const EMismatchedSenderRecipient: u64 = 0;
+const EMismatchedExchangeObject: u64 = 1;
+```
+
+#### Public Functions
+
+##### `create`
+
+This function is used to create a new escrow object. It takes four arguments: the object to be escrowed, the ID of the key that opens the lock on the object sender wants from the recipient, the intended recipient, and the transaction context.
+
+```move
+public fun create<T: key + store>(
+   escrowed: T,
+   exchange_key: ID,
+   recipient: address,
+   ctx: &mut TxContext
+) {
+   let escrow = Escrow {
+       id: object::new(ctx),
+       sender: tx_context::sender(ctx),
+       recipient,
+       exchange_key,
+       escrowed,
+   };
+
+   transfer::public_share_object(escrow);
+}
+```
+
+##### `swap`
+
+This function is used to perform the swap operation. It takes four arguments: the escrow object, the key, the locked object, and the transaction context.
+
+```move
+public fun swap<T: key + store, U: key + store>(
+   escrow: Escrow<T>,
+   key: Key,
+   locked: Locked<U>,
+   ctx: &TxContext,
+): T {
+   let Escrow {
+       id,
+       sender,
+       recipient,
+       exchange_key,
+       escrowed,
+   } = escrow;
+
+   assert!(recipient == tx_context::sender(ctx), EMismatchedSenderRecipient);
+   assert!(exchange_key == object::id(&key), EMismatchedExchangeObject);
+
+   // Do the actual swap
+   transfer::public_transfer(lock::unlock(locked, key), sender);
+   object::delete(id);
```

Comment:
This is kind of special (new) because it relies on shared object deletion -- let's call that out.